_includes/devdoc/guide_dash_features.md

```diff
@@ -109,6 +109,100 @@ had already been confirmed to a block depth of 5 in the blockchain.
+* Removed: Limit on number of inputs
+* Removed: Limit on transaction value
+* Removed: Timeout for lock - transaction locks will only be removed once the transaction is confirmed in a ChainLocked block
+* Removed: Custom InstantSend fee
+
+Note: A transaction will __not__ be included in the block template (from `getblocktemplate`) unless it:
+
+ 1. Has been locked, or
+ 2. Has been in the mempool for >=10 minutes (`WAIT_FOR_ISLOCK_TIMEOUT`)
+
+A miner may still include any transaction, but blocks containing only locked
+transactions (or ones older than the timeout) should achieve a ChainLock faster.
+This is desirable to miners since it prevents any reorgs that might orphan their
+block.
+
+*InstantSend Data Flow*
+
+| **InstantSend Client** | **Direction**  | **Peers**   | **Description** |
+| `tx` message                | → |                         | Client sends InstantSend transaction
+| **LLMQ Signing Sessions**   |   |                         | Quorums internally process locking |
+|                             |   |                         | Quorum(s) responsible for the transaction's inputs lock the inputs via LLMQ signing sessions
+|                             |   |                         | Once all inputs are locked, the quorum responsible for the overall transaction creates the transaction lock (`islock`) via an LLMQ signing session
+| **LLMQ Results**             |   |                         | Quorum results broadcast to the network |
+|                             | ← | `inv` message (islock)  | Quorum responds with lock inventory
+| `getdata` message (islock)  | → |                         | Client requests lock message
+|                             | ← | `islock` message        | Quorum responds with lock message
+
+{% endautocrossref %}
+
+### ChainLocks
+{% include helpers/subhead-links.md %}
+
+{% autocrossref %}
+
+Dash's ChainLock feature leverages [LLMQ Signing Requests/Sessions](#llmq-signing-session)
+to reduce uncertainty when receiving funds and remove the possibility of 51%
+mining attacks.
+
+For each block, an LLMQ of a few hundred masternodes (300-400) is selected and each
+participating member signs the first block that it sees extending the active
+chain at the current height. If enough members (at least 240) see the same block
+as the first block, they will be able to create a `clsig` message and propagate
+it to all nodes in the network.
+
+If a valid `clsig` message is received by a node, it must reject all blocks (and
+any descendants) at the same height that do not match the block specified in the
+`clsig` message. This makes the decision on the active chain quick, easy and
+unambiguous. It also makes reorganizations below this block impossible.
+
+When LLMQ-based InstantSend is enabled, a ChainLock is only attempted once all
+transactions in the block are locked via InstantSend. If a block contains
+unlocked transactions, retroactive InstantSend locks are established prior to
```

Comment:
Are these retroactive locks in 0.14? I remember some talk about it not making it in.

Reply:
Initially I think there was some question of whether or not they would be in it, but they were added in https://github.com/dashpay/dash/pull/2770.